### DISCUSSION TITLE
sergen transform has failed for most forms.

### DISCUSSION THREAD

**Discussion by petebrieso** [serenity-is/Serenity#6504] [Help] [answered]
In my established project (10 months old) a strange new error has suddenly occurred.
First noticed in `UserForm.ts`  (in a deployed project) it was not creating the form correctly. A little digging showed that most of the generated file `ServerTypings\Administration.UserForm.ts` was missing. 
It looks like this
```
namespace MyProject.Administration {
    export interface UserForm {
    }

    export class UserForm extends Serenity.PrefixedContext {
        static formKey = 'Administration.User';
    }
}
```
Notice all the editors are missing. I then notice the same was true for 90% of the other forms in that folder.
My local project was still building fine, so I tried a fresh `git clone` and then was able to reproduce it, BUT my colleague also did a fresh copy and it built correctly for him.
I tried updating sergen to latest `6.0.8.0`
If I manually transform with `sergen t` the same thing happens. 
Same if I deleted the generated code first.
I have 2 copies of the project open on my machine, one transforms correctly and one does not.
All deployed versions suddenly have started doing the same thing.

Please help, this is a show-stopping problem.
Project is updated to net6 all dependencies are updated.

**Comment by petebrieso** [comment, marked as answer]
Found the answer, I think. I had early on added `ServerTypings` to the `.gitignore` list which seemed to be fine, by my guess is there's a form that gets used in several places which, if not yet transformed, will break the whole transform process. So I ended up with half-transformed files, not enough to break the build but just enough to bring run-time errors as the forms were not properly created - specifically the editors were missing, so reference to fields on the form would cause an exception.

**Reply by rymk14** [reply]
hi @petebrieso  thanks for sharing. this is helpful.

To fix did you have to start checking in all of the `ServerTypings` to source control? 

We are evaluating if we can stop committing all of these autogenerated Serenity files into git or not. Current we are commiting all wwwroot/esm/* files and all Modules/ServerTypes/* files.  

Merge conflicts are starting to be a pain though, so we'd like to stop checking everything into source. And it sounds like you discovered /ServerTypes/ files might be required to check-in?

thanks for your feedback!

**Reply by volkanceylan** [reply]
ServerTypes should be check in, wwwroot/esm not, as they are build artifacts, just like under bin
